File: src/main/java/com/vc/easy/L108.scala
package com.vc.easy


class TreeNode108(var _value: Int) {
  var value: Int = _value
  var left: TreeNode108 = null
  var right: TreeNode108 = null
}

object L108 {
  def sortedArrayToBST(nums: Array[Int]): TreeNode108 = {
    def solve(lo: Int, hi: Int): TreeNode108 = {
      if(lo > hi) null
      else {
        val mid = lo + (hi - lo) / 2
        val tree = new TreeNode108(nums(mid))
        tree.left = solve(lo, mid - 1)
        tree.right = solve(mid + 1, hi)
        tree
      }
    }
    solve(0, nums.length - 1)
  }
}

File: src/main/java/com/vc/easy/L108.java
package com.vc.easy;

import com.vc.hard.TreeNode;

class L108 {
    public TreeNode sortedArrayToBST(int[] nums) {
        if(nums == null || nums.length == 0) return null;
        return helper(nums, 0, nums.length - 1);
    }

    private TreeNode helper(int[] nums, int lo, int hi) {
        if(lo > hi) return null;
        int mid = lo + (hi - lo) / 2;
        TreeNode root = new TreeNode(nums[mid]);
        root.left = helper(nums, lo, mid - 1);
        root.right = helper(nums, mid + 1, hi);
        return root;
    }
}
